### PR TITLE
add other to doc category selector ui

### PR DESCRIPTION
**Pull request by tglaz**
### Client Docs - Add "Other" Folder & Category
https://betterangels.atlassian.net/browse/DEV-829

## Summary by Sourcery

New Features:
- Add a new 'Other' category to the document upload modal, allowing users to upload documents that do not fit into existing categories.

### CODE REVIEW

File: libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@
     ConsentForm: [],
     HmisForm: [],
     IncomeForm: [],
+    OtherClientDocument: [],
```
[the remark this answers is not in the document]
Comment (tglaz) [reply libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx:37]: i may actually do that

File: libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx [hunk 2]
```diff
@@ -474,6 +490,46 @@ export default function UploadModal(props: IUploadModalProps) {
                 />
               </View>
             </View>
+
+            <View style={{ gap: Spacings.xs, marginBottom: Spacings.lg }}>
```
[the remark this answers is not in the document]
Comment (tglaz) [reply libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx:494]: i'll take a look

**Review by mikefeldberg — APPROVED** [review]
I found a pretty substantial bug that affects all of our uploads, but it's not specific to this ticket so it't not a blocker to merging

File: libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx [hunk 3]
```diff
@@ -34,6 +34,7 @@ export default function UploadModal(props: IUploadModalProps) {
     ConsentForm: [],
```
[the remark this answers is not in the document]
Comment (mikefeldberg) [reply libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx:34]: filename update wasn't built. It was included in the original design, but those requirements were descoped for the 9/30 deadline and never reintroduced

[the remark this answers is not in the document]
Comment (tglaz) [reply libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx:34]: Not sure if we're talking the same thing, but I was able to rename the file on the initial upload (changed it to `Hello.jpg`

**Review by shiozakiyuko — APPROVED** [review]

**Pull request merged**